js-apps/meep-frontend/src/js/containers/exec/scenario-update-event-pane.js

```diff
@@ -0,0 +1,428 @@
+ */
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Select } from '@rmwc/select';
+import { Grid, GridCell } from '@rmwc/grid';
+import { Elevation } from '@rmwc/elevation';
+// import { TextField } from '@rmwc/textfield';
+
+import CfgNetworkElementContainer from '@/js/containers/cfg/cfg-network-element-container';
+import CancelApplyPair from '@/js/components/helper-components/cancel-apply-pair';
+
+import { validateNetworkElement } from '@/js/containers/cfg/cfg-page-container';
+
+import {
+  TYPE_EXEC,
+  UE_TYPE_STR,
+  FOG_TYPE_STR,
+  DC_TYPE_STR,
+  UE_APP_TYPE_STR,
+  EDGE_APP_TYPE_STR,
+  CLOUD_APP_TYPE_STR,
+  MAP_VIEW,
+  NET_TOPOLOGY_VIEW,
+  EXEC_EVT_SU_ACTION,
+  EXEC_EVT_SU_REMOVE_ELEM_TYPE,
+  EXEC_EVT_SU_REMOVE_ELEM_NAME,
+  SCENARIO_UPDATE_ACTION_NONE,
+  SCENARIO_UPDATE_ACTION_ADD,
+  SCENARIO_UPDATE_ACTION_MODIFY,
+  SCENARIO_UPDATE_ACTION_REMOVE,
+  ELEMENT_TYPE_DC,
+  ELEMENT_TYPE_EDGE,
+  ELEMENT_TYPE_FOG,
+  ELEMENT_TYPE_UE,
+  ELEMENT_TYPE_UE_APP,
+  ELEMENT_TYPE_EDGE_APP,
+  ELEMENT_TYPE_CLOUD_APP
+} from '@/js/meep-constants';
+
+import {
+  uiExecChangeScenarioUpdateAction,
+  uiExecScenarioUpdateRemoveEleName,
+  uiExecScenarioUpdateRemoveEleType,
+  uiExecChangeSandboxCfg
+} from '@/js/state/ui';
+
+import {
+  execElemNew,
+  execElemClear,
+  execElemSetErrMsg,
+  EXEC_ELEM_MODE_NEW
+} from '@/js/state/exec';
+
+import {
+  getElemFieldVal,
+  FIELD_NAME,
+  FIELD_PARENT
+} from '@/js/util/elem-utils';
+
+import { updateObject } from '@/js/util/object-util';
+
+import { addElementToScenario, updateElementInScenario } from '@/js/util/scenario-utils';
+
+const elementTypes = [
+  ELEMENT_TYPE_UE,
+  ELEMENT_TYPE_FOG,
+  ELEMENT_TYPE_EDGE,
+  ELEMENT_TYPE_DC,
+  ELEMENT_TYPE_UE_APP,
+  ELEMENT_TYPE_EDGE_APP,
+  ELEMENT_TYPE_CLOUD_APP
+];
+
+var elementNames = [];
+
+class ScenarioUpdateEventPane extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      actionTypes: [
+        SCENARIO_UPDATE_ACTION_NONE,
+        SCENARIO_UPDATE_ACTION_ADD,
+        // SCENARIO_UPDATE_ACTION_MODIFY,
+        SCENARIO_UPDATE_ACTION_REMOVE
+      ]
+    };
+  }
+
+  componentDidMount() {
+    this.props.changeActionType(SCENARIO_UPDATE_ACTION_NONE);
+    this.props.execElemNew();
+    this.props.changeRemoveActionEleType('');
+    this.props.changeRemoveActionEleName('');
+  }
+
+  changeAction(action) {
+    this.props.changeActionType(action);
+    if (action === SCENARIO_UPDATE_ACTION_ADD) {
+      this.props.execElemNew();
+    } else if (action === SCENARIO_UPDATE_ACTION_REMOVE) {
+      this.props.changeRemoveActionEleType('');
+      this.props.changeRemoveActionEleName('');
+    }
+  }
+
+  changeElementType(elementType) {
+    this.props.changeRemoveActionEleType(elementType);
+    this.getElementNames(elementType, this.props.scenario);
+    this.props.changeRemoveActionEleName('');
+  }
+
+  getElementNames(elementName, scenario) {
```

Comment:
We should not parse the scenario here. If functionality is missing in the scenario-utils.js file then we can add it there. Otherwise, is this information you can obtain from the element table?

Reply:
Moved this functionality to scenario-utils.js